honeycomb/honeycomb.go

```diff
@@ -101,11 +111,34 @@ func (e *Exporter) ExportSpan(data *trace.SpanData) {
 	hs := honeycombSpan(data)
 	ev.Add(hs)
 
-	// Add an event field for each attribute
-	// Hrm. Seems like attributes are trying to tell us something about the type of span
+	// TODO: (akvanhar) do something about the attributes
 	// ev.Add(data.Attributes)
+
+	// We send these message events as 0 duration spans
 	for _, a := range data.MessageEvents {
-		ev.Add(a)
+		messageEv := e.Builder.NewEvent()
+		if e.ServiceName != "" {
+			messageEv.AddField("service_name", e.ServiceName)
+		}
+
+		// TODO: (akvanhar) We can't export these as attributes are not yet exported fields
+		// if len(a.attributes) != 0 {
+		// 	for key, value := range a.attributes {
+		// 		messageEv.AddField(key, value)
+		// 	}
+		// }
+
+		// TODO: (akvanhar) This doesn't actually do anything as libhoney's Add will only add exported fields
+		messageEv.Add(a)
+		messageEv.Add(Annotation{
+			// TODO (akvanhar): Once MessageEvent.Event is publically available, pass in the timestamp
+			// Timestamp: a.time,
+			// Name: a.msg
+			TraceID:       getHoneycombTraceID(data.SpanContext.TraceID.High, data.SpanContext.TraceID.Low),
+			ParentID:      fmt.Sprintf("%d", data.SpanContext.SpanID),
+			DurationMilli: 0,
```

Comment:
do we need to generate a span id too, or does it do this for us?

Reply:
It doesn't generate a spanID for us. I was wondering about this - I can't access the internal OpenTelemetry tools to get a unique spanID... and I could artificially stuff one in there - but I'm not even sure it really needs one. It's more like a `subEvent` than a real `subSpan`

Reply:
The more I think about it, the more I think that these messages shouldn't have spanIDs as they're not _really_ spans. They're more events that happened in the context of a span.